File: server/src/main/java/com/fluffy/oauth2/ui/OAuth2WebConfig.java
package com.fluffy.oauth2.ui;

import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@RequiredArgsConstructor
public class OAuth2WebConfig implements WebMvcConfigurer {

    private final OAuth2ProviderConverter oauth2ProviderConverter;

    @Override
    public void addFormatters(org.springframework.format.FormatterRegistry registry) {
        registry.addConverter(oauth2ProviderConverter);
    }
}

File: server/src/main/kotlin/com/fluffy/oauth2/ui/OAuth2WebConfig.kt
package com.fluffy.oauth2.ui

import org.springframework.context.annotation.Configuration
import org.springframework.format.FormatterRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class OAuth2WebConfig(
    private val oauth2ProviderConverter: OAuth2ProviderConverter,
) : WebMvcConfigurer {

    override fun addFormatters(registry: FormatterRegistry) {
        registry.addConverter(oauth2ProviderConverter)
    }
}